clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java

```diff
@@ -60,8 +60,13 @@ public static String getCommitId() {
     public static synchronized void registerAppInfo(String prefix, String id, Metrics metrics, long nowMs) {
         try {
             ObjectName name = new ObjectName(prefix + ":type=app-info,id=" + Sanitizer.jmxSanitize(id));
+            MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+            if (server.isRegistered(name)) {
+                log.info("App info {} for {} already exists, so skipping  a new mbean creation", prefix, id);
```

Comment:
nit: additional space after _skipping_.

Question: What is this log output under consumer's case? I think it's:

`App info kafka.consumer for transcript2_REALTIME-transcript-topic-0 already exists, so skipping a new mbean creation`

Could we change to:
`log.info("The mbean of App info: [{}], id: [{}] already exists, so skipping a new mbean creation.", prefix, id);`

Reply:
Yes , you are correct . I have modified

Reply:
Thanks. Please add tests for it.